src/test/compile-fail-fulldeps/proc-macro/attr-invalid-exprs.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// aux-build:attr-stmt-expr.rs
+// ignore-stage1
+
+//! Attributes producing expressions in invalid locations
+
+#![feature(proc_macro, stmt_expr_attributes)]
+
+extern crate attr_stmt_expr;
+use attr_stmt_expr::duplicate;
+
+fn main() {
+    let _ = #[no_output] "Hello, world!";
```

Comment:
This didn't produce an error, which is concerning.

Reply:
Nevermind, it hits later than the other one, but the UX is pretty bad:

    1:1: 1:1: expected expression, found `<eof>`

Reply:
Either that, or just re-span the parse error. That'd probably be easier.